WorkbookApps/Xamarin.Workbooks.WebAssembly/Client/runtime.ts

```diff
@@ -0,0 +1,498 @@
+        const jsString = Module.UTF8ToString(typeName)
+        return jsString
+    }
+
+    getClassForObject(obj: number): MonoClass {
+        const klass = this._getClass(obj)
+        const assembly = this._getAssemblyFromClass(klass)
+        const assemblyName = Module.UTF8ToString(this._getAssemblyName(assembly))
+        const classNamespace = Module.UTF8ToString(this._getClassNamespace(klass))
+        const className = Module.UTF8ToString(this._getClassName(klass))
+
+        const monoAssembly = new MonoAssembly(
+            assemblyName,
+            new Handle(assembly),
+            this)
+
+        return new MonoClass(
+            monoAssembly,
+            classNamespace,
+            className,
+            new Handle(klass),
+            this
+        )
+    }
+
+    loadAssembly(assemblyName: string): MonoAssembly {
+        let asmHandle = this.assemblyCache[assemblyName]
+        if (!asmHandle) {
+            asmHandle = new Handle(this._loadAssembly(assemblyName))
+
+            if (!asmHandle.value)
+                throw new Error(`Could not load assembly ${assemblyName}`)
+
+            this.assemblyCache[assemblyName] = asmHandle;
+        }
+
+        return new MonoAssembly(assemblyName, asmHandle, this)
+    }
+
+    findClass(assembly: MonoAssembly, namespace: string, name: string): MonoClass {
+        const key = `${assembly.handle}_${namespace}_${name}`
+
+        let klass = this.classCache[key]
+        if (!klass) {
+            klass = new Handle(this._findClass(assembly.handle.value, namespace, name))
+
+            if (!klass.value)
+                throw new Error(`Could not load class ${namespace}.${name} from ${assembly.name}`)
+
+            this.classCache[key] = klass
+        }
+
+        return new MonoClass(assembly, namespace, name, klass, this)
+    }
+
+    createObject(klass: MonoClass): ManagedObject {
+        const newObject = new Handle(this._createObject(klass.handle.value))
+        return new ManagedObject(klass, newObject, this)
+    }
+
+    findMethod(klass: MonoClass, name: string, argc: number): MonoMethod {
+        const key = `${klass.handle}_${name}_${argc}`
+
+        let method = this.methodCache[key]
+        if (!method) {
+            method = new Handle(this._findMethod(klass.handle.value, name, argc))
+
+            if (!method.value)
+                throw new Error(
+                    `Could not load method ${name} with ${argc} argument(s) from class ` +
+                    `${klass.namespace}.${klass.name} in ${klass.assembly.name}`)
+
+            this.methodCache[key] = method
+        }
+
+        return new MonoMethod(klass.assembly, klass, name, argc, method, this)
+    }
+
+    /**
+     * Boxes a number so it can be passed to Mono. Must be freed by the caller!
+     * @param number The number to box
+     * @returns A pointer to some memory with the integer boxed into it.
+     */
+    boxInteger(number: number): number {
```

Comment:
Investigate whether I can get rid of this and boxBoolean below—I think they are not needed anymore.